pallets/creditcoin/src/lib.rs

```diff
@@ -1206,6 +1207,61 @@ pub mod pallet {
 		#[transactional]
 		#[pallet::weight(<T as Config>::WeightInfo::register_transfer_ocw())]
 		pub fn register_funding_transfer(
+			origin: OriginFor<T>,
+			transfer_kind: LegacyTransferKind,
+			deal_order_id: DealOrderId<T::BlockNumber, T::Hash>,
+			blockchain_tx_id: ExternalTxId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let order = try_get_id!(DealOrders<T>, &deal_order_id, NonExistentDealOrder)?;
+
+			ensure!(order.terms.currency.is_placeholder(), Error::<T>::DeprecatedExtrinsic);
+
+			let (transfer_id, transfer) = Self::register_transfer_internal_legacy(
+				who,
+				order.lender_address_id,
+				order.borrower_address_id,
+				transfer_kind,
+				order.terms.amount,
+				deal_order_id,
+				blockchain_tx_id,
+			)?;
+			Self::deposit_event(Event::<T>::TransferRegistered(transfer_id, transfer));
+
+			Ok(())
+		}
+
+		#[transactional]
+		#[pallet::weight(<T as Config>::WeightInfo::register_transfer_ocw())]
+		pub fn register_repayment_transfer(
+			origin: OriginFor<T>,
+			transfer_kind: LegacyTransferKind,
+			repayment_amount: ExternalAmount,
+			deal_order_id: DealOrderId<T::BlockNumber, T::Hash>,
+			blockchain_tx_id: ExternalTxId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let order = try_get_id!(DealOrders<T>, &deal_order_id, NonExistentDealOrder)?;
+
+			let (transfer_id, transfer) = Self::register_transfer_internal_legacy(
+				who,
+				order.borrower_address_id,
+				order.lender_address_id,
+				transfer_kind,
+				repayment_amount,
+				deal_order_id,
+				blockchain_tx_id,
+			)?;
+			Self::deposit_event(Event::<T>::TransferRegistered(transfer_id, transfer));
+
+			Ok(())
+		}
+
+		#[transactional]
+		#[pallet::weight(<T as Config>::WeightInfo::register_transfer_ocw())]
+		pub fn register_funding_transfer_new(
```

Comment:
ugh, naming. maybe we stick with the legacy pattern, rename the old one and leave this one `register_funding_transfer`. Thoughts?

Reply:
Makes sense to me, since then the common case (the non-legacy path) is the more intuitive name.